src/ol/expr/expression.js

```diff
@@ -668,19 +634,29 @@ const parsers = {
  * @type ArgValidator
  */
 function withGetArgs(encoded, context) {
-  const arg = parse(encoded[1], context);
-  if (!(arg instanceof LiteralExpression)) {
-    throw new Error('Expected a literal argument for get operation');
-  }
-  if (typeof arg.value !== 'string') {
-    throw new Error('Expected a string argument for get operation');
-  }
-  context.properties.add(arg.value);
-  if (encoded.length === 3) {
-    const hint = parse(encoded[2], context);
-    return [arg, hint];
+  const args = [];
+  for (let i = 1, ii = encoded.length; i < ii; ++i) {
+    const arg = parse(encoded[i], context);
+    args.push(arg);
+    if (!(arg instanceof LiteralExpression)) {
+      throw new Error('Expected a literal argument for get operation');
+    }
+    if (i > 1) {
+      if (typeof arg.value !== 'string' && typeof arg.value !== 'number') {
+        throw new Error(
+          'Expected key or array index of a get operation to be a string or number',
+        );
+      }
+      continue;
+    }
+    if (typeof arg.value !== 'string') {
+      throw new Error(
+        'Expected the attribute name of a get operation to be a string',
+      );
+    }
+    context.properties.add(String(arg.value));
```

Comment:
We aren't leveraging the `properties` of the parsing context right now in Canvas, but here is the idea.  As we parse style expressions, we build up a lookup of all the data that is needed from a feature (feature id, geometry, properties).  Then during the evaluation phase, we could supply a minimal set of data in the evaluation context.  

Currently, we don't do this in Canvas.  Instead we add [all the feature properties](https://github.com/openlayers/openlayers/blob/72c3b7ff61921aa7b0551f3c833ea8c4e68336a3/src/ol/render/canvas/style.js#L78) to the evaluation context.  If we were passing the data around to/from workers, we might instead want to encode a subset of the data for that purpose.

In the WebGL renderer, the `properties` from the parsing context is passed to the evaluation context [and then used](https://github.com/openlayers/openlayers/blob/72c3b7ff61921aa7b0551f3c833ea8c4e68336a3/src/ol/render/canvas/style.js#L78) when building shaders and uploading attributes to the gpu.

To support working with a subset of feature properties in the Canvas renderer in the future, we want to keep track of these "property paths" that are used in style expressions.  Now that we support deeply nested paths, instead of the parsing context properties being `Set<string>` it could be `Array<Array<string>>` (where a single entry might be `['path', 'to', 'property']`.  Since deeply nested paths aren't supported in WebGL yet, this would break things on that side.  For now, since we aren't using the parsing context `properties` in Canvas, we could just store a dot delimited path to all the properties (e.g. `path.to.properties`).  This isn't the best long-term solution if people do crazy things like put dots in their property names, but it could be fine for now.

The tl;dr is that we don't want to add 'path', 'to', and 'property' to this `context.properties` set.  Instead we could add 'path.to.property' (and then later change the type when we want to fully support nested properties in both Canvas and WebGL).